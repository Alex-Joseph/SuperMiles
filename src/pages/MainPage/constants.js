export const actionTypes = {
  UPDATE_BALANCE: 'UPDATE_BALANCE',
  INSUFFICIENT_BALANCE: 'INSUFFICIENT_BALANCE',
};
