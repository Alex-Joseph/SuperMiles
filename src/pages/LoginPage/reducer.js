import { fromJS } from 'immutable';
import { actionTypes } from './constants';


export const initialState = fromJS({
  loggedIn: false,
  loginError: false,
});

export const authReducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case actionTypes.LOGIN_SUCCESS:
      return state
        .set('loginError', false)
        .set('loggedIn', true);
    case actionTypes.LOGIN_FAILURE:
      return state
        .set('loginError', true);
    case actionTypes.LOGOUT:
      return state
        .set('loggedIn', false);
    default:
      return state;
  }
};
