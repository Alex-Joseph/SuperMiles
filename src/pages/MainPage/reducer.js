import { fromJS } from 'immutable';
import { actionTypes } from './constants';


export const initialState = fromJS({
  balance: 50000,
  transaction: null,
});

export const walletReducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case actionTypes.UPDATE_BALANCE:
      return state
        .set('balance', action.balance)
        .set('transaction', action.transaction);
    default:
      return state;
  }
};
