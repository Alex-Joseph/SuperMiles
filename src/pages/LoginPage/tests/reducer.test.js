import { fromJS } from 'immutable';
import { actionTypes } from '../constants';
import { initialState, authReducer } from '../reducer';


describe('reducer', () => {
  describe('initial state', () => {
    it('has an initial state', () => {
      expect(initialState).toMatchSnapshot();
    });
  });

  describe('state after reducer', () => {
    it('changes state after loginSuccess is called', () => {
      const action = { type: actionTypes.LOGIN_SUCCESS };

      expect(authReducer(initialState, action)).toMatchSnapshot();
    });

    it('changes state after loginFailure is called', () => {
      const action = { type: actionTypes.LOGIN_FAILURE };

      expect(authReducer(initialState, action)).toMatchSnapshot();
    });

    it('changes state after logout is called', () => {
      const state = fromJS({ loggedIn: true });
      const action = { type: actionTypes.LOGOUT };

      expect(authReducer(state, action)).toMatchSnapshot();
    });
  });
});
