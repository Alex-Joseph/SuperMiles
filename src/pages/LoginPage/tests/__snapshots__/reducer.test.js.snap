// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer initial state has an initial state 1`] = `
Immutable.Map {
  "loggedIn": false,
  "loginError": false,
}
`;

exports[`reducer state after reducer changes state after loginFailure is called 1`] = `
Immutable.Map {
  "loggedIn": false,
  "loginError": true,
}
`;

exports[`reducer state after reducer changes state after loginSuccess is called 1`] = `
Immutable.Map {
  "loggedIn": true,
  "loginError": false,
}
`;

exports[`reducer state after reducer changes state after logout is called 1`] = `
Immutable.Map {
  "loggedIn": false,
}
`;
